app/controllers/subjects_controller.rb

```diff
@@ -1,8 +1,8 @@
 class SubjectsController < ApplicationController
   def show
     skip_authorization
-    @extracts = workflow.extracts.where(subject_id: subject.id)
-    @reductions = workflow.subject_reductions.where(subject_id: subject.id)
+    @extracts = if workflow.nil? then [] else workflow.extracts.where(subject_id: subject.id) end
+    @reductions = (workflow||project).subject_reductions.where(subject_id: subject.id)
```

Comment:
Minor comment: Reading this bugs me for some reason, it may be completely unfounded but the fact that we're unsure which resource scopes the `subject_reductions` on a controller route strikes me as a little hacky. Ideally the route syntax would isolate these from each other but i can see why that might not be the best for flat resource routes. 

I think i'd prefer to read a place holder variable that describes the loaded resource state here, perhaps `scoping_resource` vs the individual `project` / `workflow` methods that reflect on the incoming params to load the correct one.  In the show route you can switch redirect paths based on the class type of the `scoping_resource` as well. 

As i said for some reason this just bugs me and it may be unfounded.

Reply:
I think I understand what you're suggesting here; let me take a crack at it and see if I can come up with something that feels a little cleaner.

Reply:
How does this change look?